File: Examples/PHP/Files/DownloadFile.php
//ExStart:
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\Storage\StorageApi;
use Aspose\Storage\AsposeApp;

class Files {

	public $storageApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->storageApi = new StorageApi();
	}

	public function downloadFile() {
		// Upload file to Aspose Cloud Storage
		$fileName = "AutoFilter.xlsx";
		//Utils::uploadFile($fileName);

		$result = $this->storageApi->GetDownload($Path=$fileName, $versionId = null, $storage = null);
        $fh = fopen(realpath(__DIR__ . '/../../..'). '/Data/Output/test_cells.xlsx', 'w');
        fwrite($fh, $result);
        fclose($fh);
	}
}

$files = new Files();
$files->downloadFile();

?>
//ExEnd:

File: Examples/Ruby/Files/download_file.rb
#ExStart:
require 'aspose_storage_cloud'

class Files
    
  include AsposeStorageCloud

  def initialize
    # Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("App Key", "App SID")
    @storage_api = StorageApi.new
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end
  
  def download_file
    file_name = "testfile.txt"
    upload_file(file_name)
    
    response = @storage_api.get_download(file_name)
  end

end

files = Files.new()
puts files.download_file
#ExEnd: